src/ActualBlockEmitter.js

```diff
@@ -0,0 +1,26 @@
+const EventEmitter = require('events');
+const web3 = require('./utils/createAndUnlockWeb3');
+const logger = require('../src/config/winston');
+
+
+class ActualBlockEmitter extends EventEmitter {
+  constructor() {
+    super();
+    this.lastBlock = 0;
+  }
+
+  listen() {
+    setInterval(this.checkForNewBlock.bind(this), 5000);
```

Comment:
i suggest to save interval id in class, now you are not able to clean this interval